extensions/panache/hibernate-reactive-panache-common/runtime/src/main/java/io/quarkus/hibernate/reactive/panache/common/runtime/AbstractJpaOperations.java

```diff
@@ -0,0 +1,506 @@
+        return list(findAll(entityClass, sort));
+    }
+
+    public Multi<?> streamAll(Class<?> entityClass) {
+        return stream(findAll(entityClass));
+    }
+
+    public Multi<?> streamAll(Class<?> entityClass, Sort sort) {
+        return stream(findAll(entityClass, sort));
+    }
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    public Uni<Long> count(Class<?> entityClass) {
+        return getEntityManager()
+                .flatMap(em -> (Uni) em.createQuery("SELECT COUNT(*) FROM " + getEntityName(entityClass)).getSingleResult());
+    }
+
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    public Uni<Long> count(Class<?> entityClass, String query, Object... params) {
+        return getEntityManager()
+                .flatMap(em -> (Uni) bindParameters(em.createQuery(createCountQuery(entityClass, query, paramCount(params))),
+                        params).getSingleResult());
+    }
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    public Uni<Long> count(Class<?> entityClass, String query, Map<String, Object> params) {
+        return getEntityManager()
+                .flatMap(em -> (Uni) bindParameters(em.createQuery(createCountQuery(entityClass, query, paramCount(params))),
+                        params).getSingleResult());
+    }
+
+    public Uni<Long> count(Class<?> entityClass, String query, Parameters params) {
+        return count(entityClass, query, params.map());
+    }
+
+    public Uni<Boolean> exists(Class<?> entityClass) {
+        return count(entityClass).map(c -> c > 0);
+    }
+
+    public Uni<Boolean> exists(Class<?> entityClass, String query, Object... params) {
+        return count(entityClass, query, params).map(c -> c > 0);
+    }
+
+    public Uni<Boolean> exists(Class<?> entityClass, String query, Map<String, Object> params) {
+        return count(entityClass, query, params).map(c -> c > 0);
+    }
+
+    public Uni<Boolean> exists(Class<?> entityClass, String query, Parameters params) {
+        return count(entityClass, query, params).map(c -> c > 0);
+    }
+
+    public Uni<Long> deleteAll(Class<?> entityClass) {
+        return getEntityManager().flatMap(em -> em.createQuery("DELETE FROM " + getEntityName(entityClass)).executeUpdate()
+                .map(i -> i.longValue()));
+    }
+
+    public Uni<Boolean> deleteById(Class<?> entityClass, Object id) {
+        // Impl note : we load the entity then delete it because it's the only implementation generic enough for any model,
+        // and correct in all cases (composite key, graph of entities, ...). HQL cannot be directly used for these reasons.
+        return findById(entityClass, id)
+                .flatMap(entity -> {
+                    if (entity == null) {
+                        return Uni.createFrom().item(false);
+                    }
+                    return getEntityManager().flatMap(em -> em.remove(entity).map(v -> true));
+                });
```

Comment:
You could use ifNull and IfNotNull here:
```
findById(entityClass, id)
    .onItem().ifNotNull().transformToUni( entity -> s.remove( entity ).onItem().transform( ignore -> Boolean.TRUE ))
    .onItem().ifNull().continueWith( Boolean.FALSE );
```

Reply:
I can, but it's not clear to me that it'll be more readable :)